test_conformance/basic/test_progvar.cpp

```diff
@@ -547,12 +549,28 @@ static cl_int print_build_log(cl_program program, cl_uint num_devices,
     return error;
 }
 
-static void l_load_abilities(cl_device_id device)
+static cl_int l_load_abilities(cl_device_id device)
 {
+    cl_int status = CL_SUCCESS;
+
     l_has_half = is_extension_available(device, "cl_khr_fp16");
     l_has_double = is_extension_available(device, "cl_khr_fp64");
     l_has_cles_int64 = is_extension_available(device, "cles_khr_int64");
 
+    if (get_device_cl_version(device) >= Version(3, 0)) {
+        OpenCLCFeatures features;
+        status = get_device_cl_c_features(device, features);
+        if (status)
+        {
+            log_error("Couldn't query OpenCL C features for the device!\n");
+            return TEST_FAIL;
```

Comment:
Nitpick: I think this is mixing CL error codes (e.g. CL_SUCCESS) with test statuses (TEST_FAIL).  Could this just return the CL error code instead?

```suggestion
            return status;
```

Alternatively this could just `assert(status == CL_SUCCESS)`, which isn't great but is already used below.